gateway/webhook/webhook.go

```diff
@@ -276,8 +277,16 @@ func prepareRequestBody(req *http.Request, sourceType string, sourceListForParsi
 		body = []byte("{}") // If body is empty, set it to an empty JSON object
 	}
 
+	if !json.Valid(body) {
+		return nil, fmt.Errorf("\"%s\" is not a valid json", body)
+	}
+
 	if slices.Contains(sourceListForParsingParams, strings.ToLower(sourceType)) {
 		queryParams := req.URL.Query()
+		headerBytes, headerMarshalError := json.Marshal(req.Header)
```

Comment:
The intention is that `sourceListForParsingParams` is only a configuration & has no meaning other than that. If needed, that can be changed.

Reply:
are expecting we will always have the same whitelisting for query and header parsing?

Reply:
Kind of yes!

First thing, didn't want to break functionality as such now.
Ideally, I would've wanted to send whatever in request like query_params, headers, request method etc,. through and the downstream source transformation logic would decide if there is something needed for it. And do processing over it.
But since making such change might break something, decided to make the feature live behind an env.